src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -598,6 +605,34 @@ public function getFile()
     }
 
     /**
+     * Sets if the service must be shared or not.
+     *
+     * @param bool $shared Whether the service must be shared or not
+     *
+     * @return Definition The current instance
+     *
+     * @api
+     */
+    public function setShared($shared)
+    {
+        $this->shared = (bool) $shared;
```

Comment:
Do we really want to cast here? The docblock is explicit about the expected type.